src/odemis/driver/xt_client.py

```diff
@@ -2374,54 +2378,56 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                 # is eventually updated that the stage has moved unexpectedly. So, wait until the reported position is
                 # (1) not identical to the starting position and (2) not too far from the target position.
 
-                # The stage is not moving anymore, but we still want to wait until the position has been updated
-                # TODO: base the timeout on the stage movement time estimation (est. *10)
-                timeout = 10  # s
-                expected_end_time = time.time() + timeout  # s
-                check_pos = True
-
-                # TODO: Include rotation and tilt to check stage movement to target position. Test it on a hardware.
-                # The below check to skip the check on stage movement to target position if it includes r ot t
-                # is temporary. It will be removed once the behavior is extended to all the axes and confirmed to work
-                # on a hardware.
-                if 'r' in pos or 't' in pos:
-                    logging.debug("Position requested to move includes t (rx/tilt) or r (rz/rotation). "
-                                  "Reported position accuracy is not checked.")
-                    check_pos = False
-
                 # Get the target position in absolute coordinates
                 if rel:
                     target_pos = {ax: val + pos[ax] for ax, val in orig_pos.items()}
                 else:
-                    target_pos = pos
+                    target_pos = {ax: pos[ax] for ax, val in orig_pos.items()}
 
-                while check_pos:
+                # The stage is not moving anymore, but we still want to wait until the position has been updated
+                # TODO: base the timeout on the stage movement time estimation, needs to be checked on hardware
+                # axes_list = [ax for ax in orig_pos.keys()]
+                # timeout = estimate_stage_movement_time(
+                #     stage=self,
+                #     start_pos=orig_pos,
+                #     end_pos=target_pos,
+                #     axes=axes_list,
+                #     independent_axes=True,
+                # )
+                # wait_stage_move = timeout * 5 + 3  # s, adding extra margin
```

Comment:
I don't know if that's really necessary. We have already received info from the server that the stage finished moving. So hopefully it's still soon? Or can it really be very long to actually finish if the move was large?